File: server/models/course.js
const mongoose = require("mongoose");

const courseSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
    trim: true,
  },
  description: {
    type: String,
    required: true,
  },
  difficulty: {
    type: String,
    required: true,
    enum: ["Beginner", "Intermediate", "Advanced"],
  },
  lessons: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Lesson",
    },
  ],
  imageUrl: {
    type: String,
    required: true,
  },
  created_at: {
    type: Date,
    default: Date.now,
  },
  updated_at: {
    type: Date,
    default: Date.now,
  },
  isActive: {
    type: Boolean,
    default: true,
  },
});

courseSchema.pre("save", function (next) {
  if (this.isNew || this.isModified()) {
    this.updated_at = Date.now();
  }
  next();
});

module.exports = mongoose.model("Course", courseSchema);

File: ByteBattles.Server/src/ByteBattles.Core/Entities/Course.cs
namespace ByteBattles.Core.Entities;

/// <summary>
/// Represents a learning course in the system.
/// </summary>
public class Course
{
    public int Id { get; set; }
    
    public string Title { get; set; } = null!;
    
    public string Description { get; set; } = null!;
    
    /// <summary>
    /// Difficulty level: "Beginner", "Intermediate", "Advanced"
    /// </summary>
    public string Difficulty { get; set; } = null!;
    
    public string ImageUrl { get; set; } = null!;
    
    public bool IsActive { get; set; } = true;
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    
    // Navigation properties
    public virtual ICollection<UserCourseProgress> UserProgress { get; set; } = new List<UserCourseProgress>();
}

